clients/upssched.c

```diff
@@ -757,30 +747,39 @@ static void sendcmd(const char *cmd, const char *arg1, const char *arg2)
 
 		/* we're connected now */
 
-		ret = write(pipefd, enc, enclen);
+		ret = write(pipefd, enc, sizeof(enc));
 
 		/* if we can't send the whole thing, loop back and try again */
-		if ((ret < 1) || (ret != (ssize_t) enclen)) {
+		if ((ret < 1) || (ret != sizeof(enc))) {
```

Comment:
What was wrong with `enclen` caching the `sizeof(enc)` (IIRC)?

Also, lost casting for comparing `sizeof` to the known-positive `ret` here, so open to signed/unsigned mismatch warnings which now should be fatal (`ssize_t ret` vs. `size_t sizeof`).

Can't remember looking from phone, if `enclen` was also range-checked to fit in `SSIZE_MAX`...

Reply:
- `enclen` cashes the length of string pointed by `buf` (`enclen = strlen(buf)`). There is a check also of `enclen` against `SSIZE_MAX` which is pointless taking into account that `buf` / `enc` has a fixed length of `SMALLBUF` = 512 / (512 + 8). The result was that `enc` wasn't written but the size of string pointed by `buf` 
- sizeof(enc) has been casted prior to comparison

Reply:
Oh, that must have been a copy-paste typo, buflen elsewhere, enclen in some places.

As for SMALLBUF, pedantically, given that it is a coding-time "variable" (something people can change), and how int sizes vary (hopefully no 8/16 bit builds of NUT are relevant, but...) static analyzers wanted that resulting sizeof(stuff) checked for range.

Reply:
Note to self: The enclen vs buflen mess-up fixed in a later commit